### PR TITLE
Add FLASH_SIMULATOR to board xenvm and xenvm-qemu

### PR DESCRIPTION
**Pull request by dsemenets**

### CODE REVIEW

File: boards/arm64/xenvm/xenvm_defconfig [hunk 1]
```diff
@@ -33,9 +34,19 @@ CONFIG_MAIN_STACK_SIZE=32768
 CONFIG_IDLE_STACK_SIZE=32768
 CONFIG_ISR_STACK_SIZE=32768
 CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE=32768
-CONFIG_CMSIS_THREAD_MAX_STACK_SIZE=32768
-CONFIG_CMSIS_V2_THREAD_MAX_STACK_SIZE=32768
-CONFIG_CMSIS_V2_THREAD_DYNAMIC_STACK_SIZE=32768
-CONFIG_IPM_CONSOLE_STACK_SIZE=32768
+#CONFIG_CMSIS_THREAD_MAX_STACK_SIZE=32768
```
Comment (oleksiimoisieiev) [inline boards/arm64/xenvm/xenvm_defconfig:37]: please - no commented code

File: soc/arm64/xenvm/soc.h [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#ifndef _SOC_H
```
Comment (oleksiimoisieiev) [inline soc/arm64/xenvm/soc.h:1]: Still don't understand why do we need this header

Comment (dsemenets) [reply soc/arm64/xenvm/soc.h:1]: We need this header because we need FLASH_MAP which need this header

Comment (oleksiimoisieiev) [reply soc/arm64/xenvm/soc.h:1]: I've seached for soc.h in soc/arm64
The result is the following:
(ranger) ➜  arm64 git:(xrun) ✗ fnd soc.h
./nxp_imx/mimx8m/pinctrl_soc.h
./nxp_imx/mimx9/pinctrl_soc.h
./bcm_vk/viper/soc.h

where directory contents are: 
arm  bcm_vk  CMakeLists.txt  intel_socfpga  Kconfig  nxp_imx  nxp_layerscape  qemu_cortex_a53  qemu_virt_arm64  rockchip  xenvm

As you can see - not all socs have this header. Does it mean that qemu_cotrex_a53 for example, does not support FLASH?
There should be another way.

File: boards/arm64/xenvm/xenvm_defconfig [hunk 3]
```diff
@@ -33,9 +33,15 @@ CONFIG_MAIN_STACK_SIZE=32768
 CONFIG_IDLE_STACK_SIZE=32768
 CONFIG_ISR_STACK_SIZE=32768
 CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE=32768
-CONFIG_CMSIS_THREAD_MAX_STACK_SIZE=32768
-CONFIG_CMSIS_V2_THREAD_MAX_STACK_SIZE=32768
-CONFIG_CMSIS_V2_THREAD_DYNAMIC_STACK_SIZE=32768
-CONFIG_IPM_CONSOLE_STACK_SIZE=32768
```
Comment (lorc) [inline boards/arm64/xenvm/xenvm_defconfig:39]: Looks like spurious change. If you really need to remove all those lines - put them into separate commit.

File: boards/arm64/xenvm/xenvm_defconfig [hunk 4]
```diff
@@ -13,7 +13,7 @@ CONFIG_CONSOLE=y
 CONFIG_UART_CONSOLE=y
 
 # Enable PV hypervisor console
-CONFIG_UART_CONSOLEIO_XEN_HVC=y
+CONFIG_UART_XEN_HVC_CONSOLEIO=y
```
Comment (lorc) [inline boards/arm64/xenvm/xenvm_defconfig:16]: I know about that fix and it should go into another commit. It does nothing with virtual flash

File: boards/arm64/xenvm/xenvm-qemu_defconfig [hunk 5]
```diff
@@ -24,8 +24,15 @@ CONFIG_LOG=y
 CONFIG_LOG_MODE_MINIMAL=n
 
 CONFIG_TEST=y
-#CONFIG_PRINTK_SYNC=y
```
Comment (lorc) [inline boards/arm64/xenvm/xenvm-qemu_defconfig:27]: Should be in a separate commit, IMO

**Review by lorc — COMMENTED** [review]
Please don't add anything unrelated to the commit. Remember that commits should be atomic. Also, you really need to write proper commit description.

**Comment by lorc** [comment]
Also, there are some auto-checks failed. You need to address this. Except Manifest check.

File: boards/arm64/xenvm/xenvm-qemu.dts [hunk 6]
```diff
@@ -82,4 +83,45 @@
 		compatible = "xen,uart_hvc";
 		status = "okay";
 	};
```
Comment (firscity) [inline boards/arm64/xenvm/xenvm-qemu.dts:85]: Space before new node

File: boards/arm64/xenvm/xenvm-qemu.dts [hunk 7]
```diff
@@ -82,4 +83,45 @@
...
+		flash_sim0: flash_sim@0 {
+			compatible = "soc-nv-flash";
+			reg = <0x00000000 DT_SIZE_K(1024)>;
+			erase-block-size = <1024>;
+			write-block-size = <4>;
+		};
+	};
+};
+
+&flash_sim0 {
```
Comment (firscity) [inline boards/arm64/xenvm/xenvm-qemu.dts:101]: Do we actually need to edit previous node by handle in same dts?

File: boards/arm64/xenvm/xenvm.dts [hunk 8]
```diff
@@ -79,4 +80,45 @@
 		compatible = "xen,uart_hvc";
 		status = "okay";
 	};
```
Comment (firscity) [inline boards/arm64/xenvm/xenvm.dts:82]: Same comments as for xenvm-qemu

File: boards/arm64/xenvm/xenvm-qemu.dts [hunk 9]
```diff
@@ -82,4 +83,45 @@
...
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+/*
```
Comment (firscity) [inline boards/arm64/xenvm/xenvm-qemu.dts:106]: AFAIK, block comment should be also indented.

File: subsys/storage/flash_map/flash_map.c [hunk 10]
```diff
@@ -16,7 +16,6 @@
 #include <zephyr/storage/flash_map.h>
 #include "flash_map_priv.h"
 #include <zephyr/drivers/flash.h>
-#include <soc.h>
```
Comment (firscity) [inline subsys/storage/flash_map/flash_map.c:19]: Do we actually need changes to Zephyr mainline kernel? If so, at least it should be added in separate commit with proper explanation, I don't understand the reason of such change. It may break other SOC.

File: subsys/storage/flash_map/flash_map_layout.c [hunk 11]
```diff
@@ -15,7 +15,6 @@
 #include <zephyr/device.h>
 #include <zephyr/storage/flash_map.h>
 #include <zephyr/drivers/flash.h>
-#include <soc.h>
```
Comment (firscity) [inline subsys/storage/flash_map/flash_map_layout.c:18]: Same

**Review by firscity — CHANGES_REQUESTED** [review]
Also, please rebase changes on top of aos-wip branch.

File: subsys/storage/flash_map/flash_map.c (hunk 10 above)
Comment (dsemenets) [reply subsys/storage/flash_map/flash_map.c:19]: I don't know why this line present in those files. There are no dependency from soc.h.
Technically we can create empty soc.h for our board but why?

Comment (firscity) [reply subsys/storage/flash_map/flash_map.c:19]: I see only 2 valid approaches:
1. Create empty soc.h header to comply mainline code
2. Remove this includes from subsys with proper explanation and try to send it to mainline repo
Any of these changes should be in separate commit with explanation in commit message.

Comment (oleksiimoisieiev) [reply subsys/storage/flash_map/flash_map.c:19]: I believe this is an error of the author. Went through all of the soc.h files and see the following:
1) soc,h for different platforms have completely different definitions. Nothing common at all
2) Not all platforms have soc.h files which means this header is optional.
I think that common code is not a correct place for optional header without common interface.
I suggest to contact an author Andrzej Puzdrowski and ask him specifically. Who knows, maybe he will accept this change fast.

File: boards/arm64/xenvm/xenvm-qemu.dts (hunk 6 above)
Comment (oleksiimoisieiev) [reply boards/arm64/xenvm/xenvm-qemu.dts:79]: I still suggest to use overlays instead of changing dts. This is the way it was done in zephyr/samples/subsys/fs/littlefs. Why shouldn't we keep things this way?

**Review by firscity — APPROVED** [review]
`Reviewed-by: Dmytro Firsov <dmytro_firsov@epam.com>`

**Comment by firscity** [comment]
@dsemenets, please re-push commit with added Reviewed-by and it will be merged.
